apps/asap-server/src/handlers/research-output/index-handler.ts

```diff
@@ -28,24 +28,45 @@ export const indexResearchOutputHandler = (
       SquidexWebhookResearchOutputPayload
     >,
   ): Promise<void> => {
-    const researchOutput = await researchOutputController.fetchById(
-      event.detail.payload.id,
-    );
+    logger.debug(`Event ${event['detail-type']}`);
 
-    await algoliaIndex.saveObject({
-      ...researchOutput,
-      objectID: researchOutput.id,
-    });
+    try {
+      const researchOutput = await researchOutputController.fetchById(
+        event.detail.payload.id,
+      );
 
-    logger.info(`Saved Research-Output with ID ${researchOutput.id}`);
+      logger.debug(`Fetched research-output ${researchOutput.id}`);
+
+      await algoliaIndex.saveObject({
+        ...researchOutput,
+        objectID: researchOutput.id,
+      });
+
+      logger.debug(`Saved research-output ${researchOutput.id}`);
+    } catch (e) {
+      if (e?.output?.statusCode === 404) {
+        try {
+          await algoliaIndex.deleteObject(event.detail.payload.id);
+        } catch (err) {
```

Comment:
you can try catch the `deleteObject` if you like (its not necessary) but then you should still rethrow the error as we want the handler to retry if there is an error